### DISCUSSION TITLE
Hub not appearing in Device manager

### DISCUSSION THREAD

**Discussion by WhittyRobot1** [pybricks/support#1608] [Q&A] [answered]
Yes, the hub is powered off and the bluetooth light is flashing rainbow colors. In Device Manager, there are no devices showing warning signs under ports, Universal Serial Bus Controllers, or other devices. I've read other threads on this issue and I'm still not able to find the hub in device manager and install firmware. Thanks in advance for any help.

**Comment by laurensvalk** [comment]
Not sure if this is helpful, but we've had a few users who ultimately found that their USB cable was the issue. Does this persist if you try another cable?

Or similarly, does this cable work when coding USB with the official software?

**Reply by WhittyRobot1** [reply]
We’ve tried multiple cables that all work with the spike app. So, unfortunately, I don’t think it’s the cable. Thank you for replying though.

**Reply by dlech** [reply]
Is Pybricks Code able to see the hub in firmware update mode? It could be that the correct driver is already installed so you don't need to do anything in the *Device Manager* in Windows.

**Reply by WhittyRobot1** [reply]
No. It pybricks code doesn’t see the hub in update mode.

**Reply by dlech** [reply, marked as answer]
Does it show up in the Windows Settings under Bluetooth & Devices > Devices > Other Devices like this?

![image](https://github.com/pybricks/support/assets/963645/07121a3c-00ba-4e12-95c9-c607198a1aa5)

**Comment by WhittyRobot1** [comment]
No. I even went into to the spike app and paired it via bluetooth there (successfully) but it’s still not showing up in Bluetooth devices in settings.

**Reply by BertLindeman** [reply]
Hi,

Just for my understanding: 
Did you connect your spike hub via bluetooth to the spike app,
and was that spike hub running pybricks firmware?
My prime hub with pybricks firmware will not show up on the spike app on windows (1.3.0).

**Reply by WhittyRobot1** [reply]
No - I can’t install the Pybricks firmware. I tried to connect by simply pairing the hub while in Settings on my computer. The hub didn’t show up. I went into the spike app and it paired seamlessly but still didn’t show up in settings under Bluetooth devices.

**Reply by BertLindeman** [reply]
Ok, I understand.
No solution from me, sorry.

Bert

**Comment by WhittyRobot1** [comment]
Here are some pics. Ignore reference to old ev3. 
![IMG_2761](https://github.com/pybricks/support/assets/167585558/0bcc3918-07e7-4f15-b123-7632e2b9adae)

![IMG_2762](https://github.com/pybricks/support/assets/167585558/8f2d1b50-fd59-4fc1-855d-584d9708d4a1)

**Reply by dlech** [reply]
I wonder if it is one of the "Unknown USB Device (Device Descriptor Request Failed)" devices. That could be from a bad USB cable or bad USB port/hub like Laurens suggested. Do you have a different computer you could try (just for flashing the firmware)?

**Reply by WhittyRobot1** [reply]
It was one of those one of those the other day and firmware install failed and now I can’t get it to connect again. And yes, I tried it on another pc. I can try it on that other PC later today.

**Reply by dlech** [reply]
If you right-click on one of the unknown devices, does it let you remove the device? If it does, then you could "remove" it from Windows and then unplug it and plug it back in and see if Windows picks it up properly.

**Reply by WhittyRobot1** [reply]
Didn’t work. So frustrating! Thank you so much for trying to help me through this. My kid has had so many issues with the spike app and really wants this to work. I’ll try on the other pc later and post again. Let me know if you have other ideas. Thanks again.

**Comment by WhittyRobot1** [comment]
We finally got it to install the firmware. We used the secondary pc we tried a couple days ago and it magically worked. No idea what the issue was on the primary pc but it’s working now. Thanks so much for everyone who helped out!

**Reply by dlech** [reply]
Really glad to hear you were able to get it to work and thanks for following up with the solution that worked for you!

**Comment by WhittyRobot1** [comment]
Closing now. Thanks again.
